### PR TITLE
megapixels: 1.7.0 -> 1.8.0

### PR DESCRIPTION
**Pull request by Luflosi**
## Description of changes
https://gitlab.com/megapixels-org/Megapixels/-/tags/1.8.0
Also add myself as a maintainer, migrate the package to by-name, enable `strictDeps`, use finalAttrs and set `meta.mainProgram`.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/me/megapixels/package.nix
Comment (dotlambda) [inline pkgs/by-name/me/megapixels/package.nix:31]: Isn't
```suggestion
  pname = "megapixels";
```
the convention?

Comment (Luflosi) [reply pkgs/by-name/me/megapixels/package.nix]: From https://github.com/NixOS/nixpkgs/blob/c9dda2db28825cb690dc647b74c6cd7dde795d89/pkgs/README.md:
"The `pname` attribute _should_ be identical to the upstream package name."
But you're right, I can't follow that advice since the next sentence prohibits uppercase letters:
"The `pname` and the `version` attribute _must not_ contain uppercase letters — e.g., `"mplayer"` instead of `"MPlayer"`."
I traced it back to eb9ff07031d8b1845d6d84579d1b19481977e319 where Eelco added in 2009:
"Don’t use uppercase letters in the `name` attribute — e.g., `"mplayer-1.0rc2"` instead of `"MPlayer-1.0rc2"`."
What I'm missing though is an explanation of _why_ this is important. Any ideas?
In my personal opinion it is more important to follow the upstream name than to change the package name to conform to the convention.
There are also many packages that don't follow this rule, for example the R package or SDL.
Anyways, I changed `pname` back to lowercase.
Instead I decided to rename the package attribute name and add an alias since I couldn't find any rules regarding the package attribute name. What are your opinions on that change?

File: pkgs/top-level/aliases.nix
Comment (SuperSandro2000) [inline pkgs/top-level/aliases.nix:663]: We generally don't uppercase attrs as this makes it harder to discover the package and just guessing the name.

File: pkgs/by-name/me/megapixels/package.nix
Comment (SuperSandro2000) [reply pkgs/by-name/me/megapixels/package.nix]: > What I'm missing though is an explanation of _why_ this is important. Any ideas?

It makes it easier to guess it and it is more uniform. Also some projects are not very consistent. 

> There are also many packages that don't follow this rule, for example the R package or SDL.

Those should really be renamed but for packages like SDL that is usually a bigger undertaking.

> What are your opinions on that change?

Not sure if there are explicit rules about that but generally pname should be equal to attr unless eg we have a wrapper or some other good reason.

Comment (Luflosi) [reply pkgs/by-name/me/megapixels/package.nix]: Okay. I'll wait for @dotlambda's feedback too and then probably rename it back.

Comment (dotlambda) [reply pkgs/by-name/me/megapixels/package.nix]: > generally pname should be equal to attr

I agree.

**Comment by Luflosi** [comment]
I renamed it back.

**Review by Aleksanaa — APPROVED** [review]
Please squash your commits into one except `megapixels: add Luflosi as maintainer` and `megapixels: migrate to by-name` to make the history cleaner.

**Comment by Luflosi** [comment]
I squashed some commits. Did I do enough squashing? I'd like to keep "megapixels: move to new upstream repo" as it shows that the hash of the old version doesn't change.

**Comment by dotlambda** [comment]
I agree with @Aleksanaa that it should have been no more than two commits but didn't want to hold this up further.

**Pull request merged**
